src/cpp/client.cpp

```diff
@@ -665,6 +665,62 @@ void Client::use_tensor_ensemble_prefix(bool use_prefix)
     this->_use_tensor_prefix = use_prefix;
 }
 
+parsed_map Client::get_db_node_info(std::string address)
+{
+    std::string host = address.substr(0, address.find(":"));
+    uint64_t port = std::stoul (address.substr(address.find(":") + 1),
+                                nullptr, 0);
```

Comment:
Given that this is a user-facing function, we should do some checking on host and port to make sure we are getting a reasonable value, and if not, issue a helpful error.